src/engine/Operation.h

```diff
@@ -112,7 +112,10 @@ class Operation {
   virtual void setSelectedVariablesForSubquery(
       const std::vector<Variable>& selectedVariables) final;
 
-  RuntimeInformation& getRuntimeInfo() { return _runtimeInfo; }
+  RuntimeInformation& getRuntimeInfo() {
+    AD_CONTRACT_CHECK(_runtimeInfo);
```

Comment:
I think if we do this properly, we should consider incorporating the [Guideline Support Libarry](https://github.com/microsoft/GSL)
which has a `not_null` pointer wrapper for the semantics of `this pointer points to a single element and nullptr is not a valid value. Maybe incorporating this (best via CMake `FetchContent`) can also be a separate PR, but I think this is exactly what we want here. The we don't really need all those assertions, but of course in this concrete case it is fine.